src/cli.rs

```diff
@@ -1,102 +1,86 @@
-                ]),
-            SubCommand::with_name("check")
-                .about("Try building the project without rendering it. Checks links")
-                .args(&[
-                    Arg::with_name("drafts")
-                        .long("drafts")
-                        .takes_value(false)
-                        .help("Include drafts when loading the site"),
-                ])
-        ])
+use std::path::PathBuf;
+
+use clap::{Parser, Subcommand};
+
+#[derive(Parser)]
+#[clap(version, author, about)]
+pub struct Cli {
+    /// Directory to use as root of project
+    #[clap(short, long, default_value = ".")]
+    pub root: PathBuf,
+
+    /// Path to a config file other than config.toml in the root of project
+    #[clap(short, long)]
+    pub config: Option<PathBuf>,
+
+    #[clap(subcommand)]
+    pub command: Command,
+}
+
+#[derive(Subcommand)]
+pub enum Command {
+    /// Create a new Zola project
+    Init {
+        /// Name of the project. Will create a new directory with that name in the current directory
+        #[clap(default_value = ".")]
+        name: String,
+
+        /// Force creation of project even if directory is non-empty
+        #[clap(short, long)]
+        force: bool,
+    },
+
+    /// Deletes the output directory if there is one and builds the site
+    Build {
+        /// Force the base URL to be that value (defaults to the one in config.toml)
+        #[clap(short = 'u', long)]
+        base_url: Option<String>,
+
+        /// Outputs the generated site in the given path (by default 'public' dir in project root)
+        #[clap(short, long)]
+        output_dir: Option<PathBuf>,
+
+        /// Include drafts when loading the site
+        #[clap(long)]
+        drafts: bool,
+    },
+
+    /// Serve the site. Rebuild and reload on change automatically
+    Serve {
+        /// Interface to bind on
+        #[clap(short, long, default_value = "127.0.0.1")]
+        interface: String,
+
+        /// Which port to use
+        #[clap(short, long, default_value = "1111")]
+        port: u16,
+
+        /// Outputs assets of the generated site in the given path (by default 'public' dir in project root).
+        /// HTML/XML will be stored in memory.
+        #[clap(short, long)]
+        output_dir: Option<PathBuf>,
+
+        /// Changes the base_url
+        #[clap(short = 'u', long, default_value = "127.0.0.1")]
+        base_url: String,
+
+        /// Include drafts when loading the site
+        #[clap(long)]
+        drafts: bool,
+
+        /// Open site in the default browser
+        #[clap(short = 'O', long)]
+        open: bool,
+
+        /// Only rebuild the minimum on change - useful when working on a specific page/section
+        #[clap(short, long)]
+        fast: bool,
+    },
+
+    /// Try to build the project without rendering it. Checks links
+    Check {
+        /// Include drafts when loading the site
+        #[clap(long)]
+        drafts: bool,
+    },
```

Comment:
can we be explicit with all the default values as well? eg set `false` for drafts etc. I guess `Subcommand` relies on default but I'd like to be able to see the full set of option values at a glance

Reply:
I've looked at the documentation, but especially for bool flags this is neither elegant to write or widely accepted style to specify the default explicitly.

Reply:
oh :(
Just passing true/false to default doesn't work? It's kind of weird that non bool gets to have their default value in the struct but not bool since there are only 2 possible values...

Reply:
Did a quick check and clap protests with a "default value is meaningless for bool" error.

I'm guessing the rationale is that bools are simple on-off switches and don't require any special handling.